lib/src/native/realm_core.dart

```diff
@@ -2995,7 +2995,15 @@ void _intoRealmQueryArg(Object? value, Pointer<realm_query_arg_t> realm_query_ar
     realm_query_arg.ref.arg = allocator<realm_value_t>();
     realm_query_arg.ref.nb_args = 1;
     realm_query_arg.ref.is_list = false;
-    _intoRealmValue(value, realm_query_arg.ref.arg.ref, allocator);
+    _intoRealmValueHack(value, realm_query_arg.ref.arg.ref, allocator);
+  }
+}
+
+void _intoRealmValueHack(Object? value, realm_value realm_value, Allocator allocator) {
```

Comment:
I'm not sure if this is a temporary solution looking at the method name ending with `Hack`. If not, maybe we need better name here.

Reply:
This is a temporary solution until the c-api is extended to support geospatial types.